app/models/binary/DataSet.scala

```diff
@@ -164,8 +185,117 @@ class DataSetDAO @Inject()(sqlClient: SqlClient,
-      implicit ctx: DBAccessContext): Fox[List[DataSet]] =
+  def findAllWithSearch(isActiveOpt: Option[Boolean],
+                        isUnreported: Option[Boolean],
+                        organizationIdOpt: Option[ObjectId],
+                        folderIdOpt: Option[ObjectId],
+                        uploaderIdOpt: Option[ObjectId],
+                        searchQuery: Option[String],
+                        includeSubfolders: Boolean,
+                        limitOpt: Option[Int])(implicit ctx: DBAccessContext): Fox[List[DataSet]] =
+    for {
+      selectionPredicates <- buildSelectionPredicates(isActiveOpt,
+                                                      isUnreported,
+                                                      organizationIdOpt,
+                                                      folderIdOpt,
+                                                      uploaderIdOpt,
+                                                      searchQuery,
+                                                      includeSubfolders)
+      limitQuery = limitOpt.map(l => q"LIMIT $l").getOrElse(q"")
+      r <- run(q"SELECT $columns FROM $existingCollectionName WHERE $selectionPredicates $limitQuery".as[DatasetsRow])
+      parsed <- parseAll(r)
+    } yield parsed
+
+  def findAllCompactWithSearch(isActiveOpt: Option[Boolean],
+                               isUnreported: Option[Boolean],
+                               organizationIdOpt: Option[ObjectId],
+                               folderIdOpt: Option[ObjectId],
+                               uploaderIdOpt: Option[ObjectId],
+                               searchQuery: Option[String],
+                               requestingUserIdOpt: Option[ObjectId],
+                               includeSubfolders: Boolean,
+                               limitOpt: Option[Int])(implicit ctx: DBAccessContext): Fox[List[DatasetCompactInfo]] =
+    for {
+      selectionPredicates <- buildSelectionPredicates(isActiveOpt,
+                                                      isUnreported,
+                                                      organizationIdOpt,
+                                                      folderIdOpt,
+                                                      uploaderIdOpt,
+                                                      searchQuery,
+                                                      includeSubfolders)
+      limitQuery = limitOpt.map(l => q"LIMIT $l").getOrElse(q"")
+      query = q"""
+            SELECT
+              d._id,
+              d.name,
+              o.name,
+              d._folder,
+              d.isUsable,
+              d.displayName,
+              d.created,
+              COALESCE(
+                (
+                  (u.isAdmin AND u._organization = d._organization) OR
+                  u.isDatasetManager OR
+                  d._uploader = u._id OR
```

Comment:
We may need to revise that in the future. There needs to be a way to revoke access to a dataset even for the person that happens to have uploaded the data.

Reply:
If you agree we can just remove this from the relevant places now. In fact, to be allowed to upload (to a folder), you need to be admin/dataset manager or folder team manager already. So this change will not affect many (and there it may actually be intended)

Reply:
I created https://github.com/scalableminds/webknossos/issues/6780 – I decided not to do it right away because I need to double-check that this does not break the uploading process itself